ntb_hw_switchtec.c

```diff
@@ -543,9 +537,8 @@ static void switchtec_ntb_check_link(struct switchtec_ntb *sndev,
 	int old = sndev->link_is_up;
 
 	if (msg == MSG_LINK_FORCE_DOWN) {
-		schedule_work(&sndev->link_reinit_work);
-
 		if (sndev->link_is_up) {
+			schedule_work(&sndev->link_reinit_work);
```

Comment:
I don't think I like this change. If a peer dies and comes back when the link is down, then we won't reinitialize the shared memory window as we should....

Otherwise this looks good.

Reply:
Hi Logan,
Thanks for the comment. I have canceled this change. 

In our test, we used a shell script to load the NTB and Switchtec drivers. After using the force down link, we need to add a delay between  loading ntb_hw_switchtec and ntb_transport module, otherwise the NT I/O test would be fail. is it ok?